From 2dd59630054f012a5bc4325a18f93e147aa163d3 Mon Sep 17 00:00:00 2001
From: "Stanley.Yang" <Stanley.Yang@amd.com>
Date: Fri, 13 Dec 2019 15:31:44 +0800
Subject: [PATCH 1049/1453] drm/amdkcl: fix compile error

The DC_FP_START()/DC_FP_END() should not be included by
HAVE_ASM_FPU_API_H, HAVE_ASM_FPU_API only manage head file
asm/fpu/api.h, asm/i387.h.

Change-Id: I9317864f4d73a0c76fb9cbe5c1cb17d0a0763931
Signed-off-by: Stanley.Yang <Stanley.Yang@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/display/dc/os_types.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/os_types.h b/drivers/gpu/drm/amd/display/dc/os_types.h
index d31cd45dc257..fbb7d076d8f8 100644
--- a/drivers/gpu/drm/amd/display/dc/os_types.h
+++ b/drivers/gpu/drm/amd/display/dc/os_types.h
@@ -51,8 +51,10 @@
 #define dm_error(fmt, ...) DRM_ERROR(fmt, ##__VA_ARGS__)
 
 #if defined(CONFIG_DRM_AMD_DC_DCN1_0)
-#if defined(CONFIG_X86)
 #include <asm/fpu/api.h>
+#endif
+
+#if defined(CONFIG_X86)
 #define DC_FP_START() kernel_fpu_begin()
 #define DC_FP_END() kernel_fpu_end()
 #elif defined(CONFIG_PPC64)
@@ -83,7 +85,6 @@
         } \
 }
 #endif
-#endif
 
 /*
  *
-- 
2.17.1

